client/components/theme-upgrade-modal/index.tsx

```diff
@@ -100,41 +96,35 @@ export const ThemeUpgradeModal = ( {
 	const firstThemeSoftwareSet = themeSoftwareSet?.[ 0 ];
 	const bundleSettings = useBundleSettings( firstThemeSoftwareSet?.slug );
 
-	const personalPlanProduct = useSelect(
-		( select ) => select( ProductsList.store ).getProductBySlug( PLAN_PERSONAL ),
-		[]
-	);
-	const premiumPlanProduct = useSelect(
-		( select ) => select( ProductsList.store ).getProductBySlug( 'value_bundle' ),
-		[]
-	);
-	const businessPlanProduct = useSelect(
-		( select ) => select( ProductsList.store ).getProductBySlug( 'business-bundle' ),
-		[]
-	);
-	const monthlyBusinessPlanProduct = useSelect(
-		( select ) => select( ProductsList.store ).getProductBySlug( 'business-bundle-monthly' ),
-		[]
-	);
-	const biAnnualBusinessPlanProduct = useSelect(
-		( select ) => select( ProductsList.store ).getProductBySlug( 'business-bundle-2y' ),
-		[]
-	);
-	const triAnnualBusinessPlanProduct = useSelect(
-		( select ) => select( ProductsList.store ).getProductBySlug( 'business-bundle-3y' ),
-		[]
+	const requiredPlanProduct = useSelect(
+		( select ) => select( ProductsList.store ).getProductBySlug( requiredPlan ),
+		[ requiredPlan ]
 	);
 
 	//Wait until we have theme and product data to show content
-	const isLoading = ! premiumPlanProduct || ! businessPlanProduct || ! theme.data;
+	const isLoading = ! requiredPlanProduct || ! theme.data;
 
 	const personalPlanName = getPlan( PLAN_PERSONAL )?.getTitle() || '';
 	const premiumPlanName = getPlan( PLAN_PREMIUM )?.getTitle() || '';
 	const businessPlanName = getPlan( PLAN_BUSINESS )?.getTitle() || '';
 	const ecommercePlanName = getPlan( PLAN_ECOMMERCE )?.getTitle() || '';
 
+	const getPlanTextByTerm = ( term: string, cost: string ) => {
```

Comment:
Dissatisfied with this, but when I looked into it earlier (when using a different approach) I got into a rabbit hole. It's no worse than the current live implementation...